spring-boot/src/main/java/org/springframework/boot/jta/narayana/NarayanaRecoveryManagerBean.java

```diff
@@ -41,8 +40,8 @@ public NarayanaRecoveryManagerBean(RecoveryManagerService recoveryManagerService
 		this.recoveryManagerService = recoveryManagerService;
 	}
 
-	@Override
-	public void afterPropertiesSet() throws Exception {
+	@EventListener
+	public void create(ApplicationReadyEvent ignored) throws Exception {
```

Comment:
A user is free to do pretty much whatever they want during application startup. Beyond and application's own logic, there's also technologies like Flyway, Liquibase, and Hibernate to consider. I don't think Flyway or Liquibase will ever use JTA transactions but Hibernate definitely can. Hibernate will access the database during startup and I believe it may use transactions when it does so. Spring Data JPA also makes calls to Hibernate during its initialisation. I"m not sure if those calls are transactional or not but it needs to be considered here.